### PR TITLE
Ruby: Credentials bugs fix

### PR DESCRIPTION
**Pull request by landrito**
This also moves the credentials class to the top level scope since the scopes are not per interface but rather a property of the model.

Monitoring unit and smoke tests now pass with these changes.

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceTransformer.java [hunk 1]
```diff
@@ -419,6 +409,11 @@ private String topLevelPackagePath(SurfaceNamer namer) {
     return Joiner.on(File.separator).join(paths);
   }
 
+  private String topLevelNamespace(SurfaceNamer namer) {
```
Comment (geigerj) [inline src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceTransformer.java]: There are several places in this file (e.g., `generateTopLevelIndexView`) where we're calling `x.subList(0, x.size() - 1)`, seemingly all to get the topLevelNamespace. This logic should be consolidated.

File: src/main/java/com/google/api/codegen/transformer/ruby/RubySurfaceNamer.java [hunk 2]
```diff
@@ -332,6 +332,19 @@ public String getTopLevelIndexFileImportName() {
     return Joiner.on(File.separator).join(newNames.toArray());
   }
 
+  @Override
+  public String getCredentialsClassImportName() {
+    List<String> parts = getApiModules();
+    // Place credentials in top-level namespace.
+    parts = parts.subList(0, parts.size() - 1);
```
Comment (geigerj) [inline src/main/java/com/google/api/codegen/transformer/ruby/RubySurfaceNamer.java]: Another instance of popping the last module that ideally should be consolidated into common logic.

File: src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceTransformer.java (hunk 1 above)
Comment (landrito) [inline src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceTransformer.java]: Done.

File: src/main/java/com/google/api/codegen/transformer/ruby/RubySurfaceNamer.java (hunk 2 above)
Comment (landrito) [inline src/main/java/com/google/api/codegen/transformer/ruby/RubySurfaceNamer.java]: Done.

**Pull request merged**